app/scripts/lib/AbstractPetnamesBridge.ts

```diff
@@ -0,0 +1,240 @@
+   */
+  #getPetnameEntries(): PetnameEntry[] {
+    const { names } = this.#nameController.state;
+    const entries: PetnameEntry[] = [];
+    for (const type of Object.values(NameType)) {
+      for (const variation of Object.keys(names[type])) {
+        for (const value of Object.keys(names[type][variation])) {
+          const entry = names[type][variation][value];
+          entries.push({
+            value,
+            type,
+            name: entry.name,
+            sourceId: entry.sourceId,
+            variation,
+          } as SetNameRequest);
+        }
+      }
+    }
+    return entries;
+  }
+
+  /**
+   * Updates Petnames with the given entry.
+   *
+   * @param type - The type of change that occurred.
+   * @param entry - The entry to update the name controller with.
+   */
+  #updatePetnameEntry(type: ChangeType, entry: PetnameEntry): void {
+    if (type === ChangeType.DELETED) {
+      this.#nameController.setName({ ...entry, name: null });
+    } else {
+      // ADDED or UPDATED
+      this.#nameController.setName(entry);
+    }
+  }
+
+  /**
+   * Get a string key for the given entry.
+   *
+   * @param entry
+   */
+  #getKey(entry: PetnameEntry): string {
+    return `${entry.type}/${entry.variation}/${entry.value}`;
+  }
+
+  /**
+   * Computes the list of changes between the previous and new entries.
+   *
+   * @param prevEntries - The previous entries.
+   * @param newEntries - The new entries.
+   * @returns A ChangeList object representing the changes that occurred between prevEntries and newEntries.
+   */
+  #computeChangeList(
+    prevEntries: PetnameEntry[],
+    newEntries: PetnameEntry[],
+  ): ChangeList {
+    const added: PetnameEntry[] = [];
+    const updated: PetnameEntry[] = [];
+    const deleted: PetnameEntry[] = [];
+
+    const prevEntriesMap = new Map(
+      prevEntries.map((e) => [this.#getKey(e), e]),
+    );
+    const newEntriesMap = new Map(newEntries.map((e) => [this.#getKey(e), e]));
+
+    newEntriesMap.forEach((newEntry, newKey) => {
+      const oldEntry = prevEntriesMap.get(newKey);
+      if (oldEntry) {
+        if (newEntry.name !== oldEntry.name) {
+          updated.push(newEntry);
+        }
+      } else {
+        added.push(newEntry);
+      }
+    });
+
+    if (!this.#isTwoWay) {
+      prevEntriesMap.forEach((oldEntry, oldKey) => {
+        if (!newEntriesMap.has(oldKey)) {
+          deleted.push(oldEntry);
+        }
+      });
+    }
+
+    return {
+      [ChangeType.ADDED]: added,
+      [ChangeType.UPDATED]: updated,
+      [ChangeType.DELETED]: deleted,
+    };
+  }
+
+  /**
+   * Applies the given change list to either the Petnames or Source, depending on the current synchrnoization direction.
+   *
+   * @param changeList
+   */
+  #applyChangeList(changeList: ChangeList): void {
+    const applyChange =
+      this.#synchronizingDirection === 'Source->Petnames'
+        ? this.updateSourceEntry.bind(this)
```

Comment:
Am I missing something or is this ternary the wrong way round?

Reply:
Good catch. Done in next update.